src/StoreKeeper/WooCommerce/B2C/PaymentGateway/PaymentGateway.php

```diff
@@ -467,6 +465,28 @@ public function addGatewayClasses($default_gateway_classes)
         return array_merge($default_gateway_classes, $gateway_classes);
     }
 
+    protected function checkRefundCreationAllowed(array $order, float $refundAmount, $orderId, $storeKeeperOrderId): bool
+    {
+        if ($this->storekeeperOrderHasRefundWithReturnPayment($order)) {
+            $paidValue = $order['paid_value_wt'];
+            $refundedValue = $order['paid_back_value_wt'];
+            $maxRefundValue = $paidValue - $refundedValue;
+            $floatEpsilon = 0.00001;
+            // Epsilon is only defined for PHP 7.2 and above
+            if (defined('PHP_FLOAT_EPSILON')) {
+                $floatEpsilon = PHP_FLOAT_EPSILON;
+            }
+
+            if ($maxRefundValue <= 0 || abs($refundAmount - $maxRefundValue) < $floatEpsilon) {
+                LoggerFactory::create('refund')->error('Order has refund on BackOffice already and refund is more than the expected amount', ['order_id' => $orderId, 'storekeeper_id' => $storeKeeperOrderId]);
```

Comment:
maybe move the logger outside this function? than less params and checking function does not log errors

Reply:
Moved logger after method call.